integrations/azure-devops/azure_devops/webhooks/listeners/push.py

```diff
@@ -27,19 +29,33 @@ async def on_hook(self, data: Dict[str, Any]) -> None:
 
         ref_update_tasks = []
         for update in updates:
-            task = asyncio.create_task(self.process_ref_update(config, update))
+            task = asyncio.create_task(self.process_ref_update(config, update, data))
             ref_update_tasks.append(task)
 
+        logger.debug(f"Created {len(ref_update_tasks)} tasks for processing updates")
+
         await asyncio.gather(*ref_update_tasks, self.register_repository(push_data))
 
     async def process_ref_update(
-        self, config: GitPortAppConfig, update: Dict[str, Any]
+        self,
+        config: GitPortAppConfig,
+        update: Dict[str, Any],
+        initial_data: dict[str, Any] = {},
```

Comment:
this is unclear, what is `initial_data`, you are relying on that you'll always have `resource` -> `repository` -> `defaultBranch` without ever specifying the structure of the object.

this makes it very hard to maintain.

I think the logic of whether to even process the event should be outside of the `process_ref_update` and the `process_ref_update` should only do what its means which is to process the update

Reply:
having the validation logic prior to spawning the task will reduce a lot of redundant tasks that will be spawned and closed right a way

Reply:
I understand what you are saying here. `initial_data` is the initial data gotten from the webhook event itself. The payload contains the default branch information for that event. It would be time consuming and redundant to map out the entirety of that object, plus it can change at anytime. The logic of whether to process the event was already in place and I didn't want to introduce significant changes to the code.

However I can do the following:
- Validate, extract `defaultBranch`, and pass it to the `process_ref_update` method
- Since you mentioned it, I will extract the logic for checking whether the event should be processed so the `process_ref_update` method does only one thing.
- Add comments explaining the existence of `resource -> repository -> defaultBranch`

What do you think?